cli/bench_command.go

```diff
@@ -66,305 +62,540 @@ type benchCmd struct {
+		f.Flag("storage", "JetStream storage (memory/file) for the \"benchstream\" stream").Default("file").EnumVar(&c.storage, "memory", "file")
+		f.Flag("replicas", "Number of stream replicas for the \"benchstream\" stream").Default("1").IntVar(&c.replicas)
+		f.Flag("maxbytes", "The maximum size of the stream or KV bucket in bytes").Default("1GB").StringVar(&c.streamMaxBytesString)
+		f.Flag("retries", "The maximum number of retries in JS operations").Default("3").IntVar(&c.retries)
+		f.Flag("history", "History depth for the bucket in KV mode").Default("1").Uint8Var(&c.history)
+	}
+
+	benchCommand := app.Command("bench", "Benchmark utility")
+	addCheat("bench", benchCommand)
+
+	benchCommand.HelpLong(benchHelp)
+
+	corePub := benchCommand.Command("pub", "Publish Core NATS messages").Action(c.pubAction)
+	corePub.Arg("subject", "Subject to use for the benchmark").Required().StringVar(&c.subject)
+	addCommonFlags(corePub)
+	addPubFlags(corePub)
+
+	coreSub := benchCommand.Command("sub", "Subscribe to Core NATS messages").Action(c.subAction)
+	coreSub.Arg("subject", "Subject to use for the benchmark").Required().StringVar(&c.subject)
+	coreSub.Flag("multisubject", "Multi-subject mode, each message is published on a subject that includes the publisher's message sequence number as a token").UnNegatableBoolVar(&c.multiSubject)
+	addCommonFlags(coreSub)
+
+	microService := benchCommand.Command("service", "Micro-service mode")
+
+	request := microService.Command("request", "Send requests and wait for a reply").Action(c.requestAction)
+	request.Arg("subject", "Subject to use for the benchmark").Required().StringVar(&c.subject)
+	// TODO: support randomized payload data
+	request.Flag("sleep", "Sleep for the specified interval between requests").Default("0s").DurationVar(&c.sleep)
+	addCommonFlags(request)
+
+	reply := microService.Command("serve", "Serve requests").Action(c.serveAction)
+	reply.Arg("subject", "Subject to use for the benchmark").Required().StringVar(&c.subject)
+	reply.Flag("sleep", "Sleep for the specified interval before replying to the request").Default("0s").DurationVar(&c.sleep)
+	addCommonFlags(reply)
+
+	jsCommand := benchCommand.Command("js", "JetStream benchmark commands")
+
+	jspub := jsCommand.Command("pub", "Publish JetStream messages").Action(c.jspubAction)
+	jspub.Arg("subject", "Subject to use for the benchmark").Required().StringVar(&c.subject)
+	jspub.Flag("batch", "The number of asynchronous JS publish calls before waiting for all the publish acknowledgements (set to 1 for synchronous)").Default("100").IntVar(&c.batchSize)
+	addCommonFlags(jspub)
+	addPubFlags(jspub)
+	addJSCommonFlags(jspub)
+	addJSPubFlags(jspub)
+
+	jsOrdered := jsCommand.Command("ordered", "Consume JetStream messages from a consumer using an ephemeral ordered consumer").Action(c.jsOrderedAction)
+	jsOrdered.Flag("consumer", "Specify the durable consumer name to use").Default(DefaultDurableConsumerName).StringVar(&c.consumerName)
```

Comment:
above you say '...using an ephemeral' here you say durable consumer name - i dont think ordered supports durable?

Reply:
Good catch, flag removed.